### DISCUSSION TITLE
Configuration file

### DISCUSSION THREAD

**Discussion by jvde-github** [jvde-github/AIS-catcher#40] [Ideas]
Work has started on the implementation of a function to specify the AIS-catcher commands via a configuration file instead of the command line:
```
AIS-catcher -C config.json
```
With a file format as per below (json). Any thoughts or comments on the setup, format or usefulness appreciated! 

J.

```json
{
	"config": "aiscatcher",
	"version": "1",
	"serial": "00000001",
	"input": "rtlsdr",
	"rtlsdr": {
		"active" : true,
		"rtlagc": true,
		"tuner": "auto",
		"bandwidth": "192K",
		"sample_rate": "1536K"
	},
	"udp": [{
		"host": "ais.fleetmon.com",
		"port": 0
	}, {
		"active": true,
		"host": "hub.shipxplorer.com",
		"port": 0
	}],
	"http": [{
		"url": "https://ais.chaos-consulting.de/shipin/index.php",
		"userpwd": "user:pwd",
		"interval": 30,
		"gzip": false,
		"response": false
	}, {
		"url": "http://aprs.fi/jsonais/post/secret_key",
		"id": "myid",
		"interval": 60,
		"protocol": "aprs",
		"response": false
	}]
}
```

**Comment by michaelgeorge001** [comment]
Sounds very useful. A nice feature would be a command line option to export the current command line configuration to the json config file. Unless there is already one, and I have missed it.

**Comment by jvde-github** [comment]
I like that. Thanks. Plan is still, at some point, to add functionality to edit the settings in the web interface for which I need this functionality. Need to sort out security then first. Thanks
